packages/ember-toucan-core/src/components/form/controls/combobox.ts

```diff
@@ -0,0 +1,537 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { next } from '@ember/runloop';
+import { isEqual as emberIsEqual } from '@ember/utils';
+
+import { offset, size } from '@floating-ui/dom';
+
+import OptionComponent, {
+  selector as optionComponentSelector,
+} from '../../../-private/components/form/controls/combobox/option';
+import Chevron from '../../../-private/icons/chevron';
+
+import type { Middleware as VelcroMiddleware } from '@floating-ui/dom';
+import type { WithBoundArgs } from '@glint/template';
+
+export type Option = string | Record<string, unknown> | undefined;
+
+export interface ToucanFormComboboxControlComponentSignature<
+  OPTION extends Option
+> {
+  Args: {
+    /**
+     * A CSS class to add to this component's content container.
+     * Commonly used to specify a `z-index`.
+     */
+    contentClass?: string;
+
+    /**
+     * Sets the input to an errored-state via styling.
+     */
+    hasError?: boolean;
+
+    /**
+     * Sets the `disabled` attribute of the input.
+     */
+    isDisabled?: boolean;
+
+    /**
+     * Sets the `readonly` attribute of the input.
+     */
+    isReadOnly?: boolean;
+
+    /**
+     * A string to display when there are no results after the user's filter.
+     */
+    noResultsText?: string;
+
+    /**
+     * Called when the user makes a selection.
+     * It is called with the selected option (derived from `@options`) as its only argument.
+     */
+    onChange?: (option: unknown) => void;
+
+    /**
+     * The function called when a user types into the combobox textbox, typically used to write custom filtering logic.
+     */
+    onFilter?: (input: string) => OPTION[];
```

Comment:
I saw we are calling this with ´await`, so should be this?

```suggestion
    onFilter?: (input: string) => OPTION[] | Promise<OPTION[]>;
```

Reply:
Ah yeah, I backed out of it being a promise for now due as the current implementation is not promise based.  Let me revert the `await`!  We can add that support back if we find out folks need it! https://github.com/CrowdStrike/ember-toucan-core/pull/200/commits/d036a432a1a36b293204e16f34cc57f437185cb5